src/test/scala/com/microsoft/hyperspace/TestUtils.scala

```diff
@@ -24,4 +26,24 @@ object TestUtils {
     result.state = state
     result
   }
+
+  /**
+   * Split path into its segments and return segment names as a sequence.
+   * For e.g. a path "file:/C:/d1/d2/d3/f1.parquet" will return
+   * Seq("file:/C:/", "d1", "d2", "d3", "f1.parquet")
+   *
+   * @param path Path to split into segments.
+   * @return Segments as a seq.
+   */
+  def splitPath(path: fs.Path): Seq[String] = {
+    var initialPath = path
+    var splits = Seq[String]()
+    while (initialPath.getParent != null) {
+      splits :+= initialPath.getName
+      initialPath = initialPath.getParent
+    }
+    // Initial path is now root. It's getName returns "" but toString returns actual path,
+    // E.g. "file:/C:/".
+    splits :+ initialPath.toString
+  }
```

Comment:
Are you implementing the following (which should have the same order as the comment)?
```scala
  def splitPath(path: Path): Seq[String] = {
    if (path.getParent == null) {
      Seq(path.toString)
    } else {
      splitPath(path.getParent) :+ path.getName
    }
  }
```

Reply:
thanks @imback82 , I was relying on the reverse order of what I wrote in the comments. Thanks for the catch.

Reply:
fixed the comments section to match the existing implementation.